stdlib/internal/promql/join.md

```diff
@@ -0,0 +1,91 @@
+### join
+
+`join` merges two input streams into a single output stream.
+Records that have the same group key and `_time` values will be joined in the output.
+The output groups will be the same as the input groups.
+
+`join` has the following properties:
+
+| Name    | Type     | Description |
+| ------- | -------- | ----------- |
+| tables  | object   | Specifies the two table streams to join. |
+| fn      | function | Defines the function that joins the rows of each table. The return value is an object which defines the output record structure. |
+
+Ex.
+```
+import "internal/promql"
+
+left = from(bucket: "my-bucket")
+    |> range(start: -1h)
+    |> filter(fn: (r) => r._field == "A")
+    |> drop(columns: ["_field"])
+    |> rename(columns: {_value: "a"})
+
+right = from(bucket: "my-bucket")
+    |> range(start: -1h)
+    |> filter(fn: (r) => r._field == "B")
+    |> drop(columns: ["_field"])
+    |> rename(columns: {_value: "b"})
+
+promql.join(tables: {left, right}, fn: (t) => {
```

Comment:
is `{left, right}`  implicit syntax for object literals?   I do like that syntax but should we be more explicit about what's happening here?  maybe name the streams `t1`  and `t2`?

Reply:
Yes that's sugar for `{left: left, right: right}`. I can be more explicit.